CMakeLists.txt

```diff
@@ -20,6 +20,41 @@ if(NOT Poco_FOUND)
       -DPOCO_UNBUNDLED:BOOL=ON
       -DCMAKE_C_FLAGS="-Wno-shift-negative-value")
   endif()
+  if(DEFINED CMAKE_TOOLCHAIN_FILE)
+    list(APPEND extra_cmake_args "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}")
+    if(ANDROID)
+      if(DEFINED ANDROID_ABI)
+        list(APPEND extra_cmake_args "-DANDROID_ABI=${ANDROID_ABI}")
+      endif()
+      if(DEFINED ANDROID_CPP_FEATURES)
+        list(APPEND extra_cmake_args "-DANDROID_CPP_FEATURES=${ANDROID_CPP_FEATURES}")
+      endif()
+      if(DEFINED ANDROID_FUNCTION_LEVEL_LINKING)
+        list(APPEND extra_cmake_args "-DANDROID_FUNCTION_LEVEL_LINKING=${ANDROID_FUNCTION_LEVEL_LINKING}")
+      endif()
+      if(DEFINED ANDROID_NATIVE_API_LEVEL)
+        list(APPEND extra_cmake_args "-DANDROID_NATIVE_API_LEVEL=${ANDROID_NATIVE_API_LEVEL}")
+      endif()
+      if(DEFINED ANDROID_NDK)
+        list(APPEND extra_cmake_args "-DANDROID_NDK=${ANDROID_NDK}")
+      endif()
+      if(DEFINED ANDROID_STL)
+        list(APPEND extra_cmake_args "-DANDROID_STL=${ANDROID_STL}")
+      endif()
+      if(DEFINED ANDROID_TOOLCHAIN_NAME)
+        list(APPEND extra_cmake_args "-DANDROID_TOOLCHAIN_NAME=${ANDROID_TOOLCHAIN_NAME}")
+      endif()
+      # TODO(esteve): replace with list(FILTER ...) when switching to CMake 3.7
+      set(extra_cmake_args_tmp)
+      foreach(extra_cmake_arg ${extra_cmake_args})
+        if(NOT extra_cmake_arg STREQUAL "-DPOCO_UNBUNDLED:BOOL=ON")
+          list(APPEND extra_cmake_args_tmp "${extra_cmake_arg}")
+        endif()
+      endforeach()
+      set(extra_cmake_args "${extra_cmake_args_tmp}")
+      list(APPEND extra_cmake_args -DPOCO_UNBUNDLED:BOOL=OFF)
```

Comment:
The whole block from line 47 to 55 can be removed since the default value of `POCO_UNBUNDLED` is `OFF` anyway. That is why we had to pass with `ON` in the first place.